src/main/java/io/xpring/Utils.java

```diff
@@ -68,6 +68,37 @@ public static String byteArrayToHex(byte [] bytes) {
         return data;
     }
 
+    /**
+     * Encode the given classic address and tag into an X-Address.
+     *
+     * @param classicAddress A classic address to encode.
+     * @param tag An optional tag to encode.
+     * @return A new X-Address if inputs were valid, otherwise undefined.
+     * @see <a href="https://xrpaddress.info/">https://xrpaddress.info/</a>
+     */
+    public static String encodeXAddress(String classicAddress, Long tag) {
+        try {
+            return javaScriptUtils.encodeXAddress(classicAddress, tag);
+        } catch (Exception exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    /**
+     * Decode a {@link ClassicAddress} from a given X-Address.
+     *
+     * @param xAddress The xAddress to decode.
+     * @return A {@link ClassicAddress} if the inputs were valid, otherwise null.
+     * @see <a href="https://xrpaddress.info/">https://xrpaddress.info/</a>
+     */
+    public static ClassicAddress decodeXAddress(String xAddress) {
+        try {
+            return javaScriptUtils.decodeXAddress(xAddress);
+        } catch (Exception exception) {
```

Comment:
I don't see any checked exceptions in the signature of the that class. Is there another reason for wrapping this in a `RuntimeException` that I'm not understanding?

Reply:
This should never happen. Because of the nature of JavaScript we can't be sure a given function exists before running it (or that it won't throw given some inputs). As a result, there's really not much we can do.

If the JavaScript is malformed, this whole library is basically useless so we just crash. There's no recovery if the core is corrupted. The good news is that we'd never be able to pass the tests needed to deploy the library, so this is kind of just an impossible use case. 

Lastly, `Utils` is acting as a facade. You could imagine that one day we might want to replace this with either a native implementation or something else that would not throw. No need to pollute our public API.

Reply:
I think what @theotherian is saying is that you can probably just get rid of the try/catch if `javaScriptUtils.decodeXAddress(xAddress)` doesn't explicitly declare a checked-exception.